firefox-ios/firefox-ios-tests/Tests/XCUITests/FeatureFlaggedTestBase.swift

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+
+/// Provides utilities to enable experiments and feature flags for specific test cases within a test class.
+///
+/// See also:
+/// https://github.com/mozilla-mobile/firefox-ios/wiki/Automated-UI-Tests#experiments--feature-flags
+class FeatureFlaggedTestBase: BaseTestCase {
+    override func setUpApp() {
+        // Important: `app.launch()` must be called explicitly in each test case.
+        setUpLaunchArguments()
+    }
+
+    /// Adds experiment data and feature flag information to the app's launch arguments.
+    ///
+    /// - Parameters:
+    ///   - jsonFileName: The name of the JSON file containing experiment configurations.
+    ///   - featureName: The feature name as defined in the Nimbus feature YAML file,
+    ///                  written in kebab-case under the `features:` section.
+    func addLaunchArgument(jsonFileName: String, featureName: String) {
+        var launchArgs = app.launchArguments
+        launchArgs.append("\(LaunchArguments.LoadExperiment)\(jsonFileName)")
+        launchArgs.append("\(LaunchArguments.ExperimentFeatureName)\(featureName)")
+        app.launchArguments = launchArgs
+    }
+}
+
+/// Provides utilities to enable experiments and feature flags for an entire test class.
+/// Use this when *all* tests in the class require the same experiment and feature flag.
+///
+/// See also:
+/// https://github.com/mozilla-mobile/firefox-ios/wiki/Automated-UI-Tests#experiments--feature-flags
+class FeatureFlaggedTestSuite: FeatureFlaggedTestBase {
```

Comment:
@lmarceau sorry I don't get this looking at the code, I will run the PR on a branch to get more familiar, in the meatime, let me ask here... would it be possible to launch the same test with an experiment on off, wihtout manually changing the .json file?
The idea is to have for example test1experimentOn, that uses some launch arguments and same test1experimentOff, with the launch arguments to have the experiment off, the app will be launched with the UI corresponding to the status of the experiment given by launch arguments.
This way we can automatically have both tests running in main or any branch at the same time, without having to manually make changes.

Reply:
The intent was to make it easier to do like the documentation to ["Experiment for a test class"](https://github.com/mozilla-mobile/firefox-ios/wiki/Automated-UI-Tests#experiment-for-a-test-class), but this means we apply the same JSON file to the whole test class (so we cannot change it from ON to OFF and vice-versa within the same test file). I am not entirely sure if this will be useful? I can remove that code and we just keep `FeatureFlaggedTestBase` for now, and if there's a need we can add `FeatureFlaggedTestSuite` back? Let me know what you think!

Reply:
Yeah we could duplicate a test class, have it ON in one file and OFF in another file!

Reply:
that is great! that is exactly what we would need. Let us give it a try @dragosb01 @clarmso with an experiment

Reply:
Awesome! FYI I plan to add back the tab tray UI experiment we needed to turn off in my next PR (with [this task](https://mozilla-hub.atlassian.net/browse/FXIOS-12107))